msgpack-core/src/main/java/org/msgpack/value/ValueFactory.java

```diff
@@ -124,17 +119,67 @@ ImmutableMapValue newMapValue(Map<K, V> map) {
             kvs[index] = pair.getValue();
             index++;
         }
-        return newMapValue(kvs);
+        return newMap(kvs);
     }
 
-    public static ImmutableMapValue newMapValue(Value[] kvs) {
+    public static ImmutableMapValue newMap(Value[] kvs) {
         if (kvs.length == 0) {
             return ImmutableMapValueImpl.empty();
         }
         return new ImmutableMapValueImpl(Arrays.copyOf(kvs, kvs.length));
     }
 
-    public static ImmutableExtendedValue newExtendedValue(byte type, byte[] data) {
-        return new ImmutableExtendedValueImpl(type, data);
+    public static ImmutableMapValue emptyMap() {
+        return ImmutableMapValueImpl.empty();
+    }
+
+
+    public static class MapEntry {
+        public final Value key;
+        public final Value value;
+
+        public MapEntry(Value key, Value value) {
+            this.key = key;
+            this.value = value;
+        }
+    }
+
+    public static MapValue newMap(MapEntry... pairs) {
+        MapBuilder b = new MapBuilder();
+        for(MapEntry p : pairs) {
+            b.put(p);
+        }
+        return b.build();
+    }
+
+
+    public static MapBuilder newMapBuilder() {
+        return new MapBuilder();
+    }
+
+    public static MapEntry newMapEntry(Value key, Value value) {
+        return new MapEntry(key, value);
+
+    }
+
+    public static class MapBuilder {
+        private Map<Value, Value> map = new HashMap<Value, Value>();
+        public MapBuilder() {}
+
```

Comment:
It's good idea to add `putAll(Iterable<? extends Map.Entry<? extends Value,? extends Value>> entries)` and `putAll(Map<? extends Value,? extends Value> map)` because Guava's ImmutableMap.Builder has those methods.